src/relay.js

```diff
@@ -107,6 +107,127 @@ export function nodeType(connectionType) {
   return connectionType._fields.edges.type.ofType._fields.node.type;
 }
 
+function orderByAttribute(orderAttr, {source, args, context, info}) {
+  return typeof orderAttr === 'function' ? orderAttr(source, args, context, info) : orderAttr;
+}
+
+function getOrder(orderBy, options) {
+  const {model} = options;
+  const result = orderBy.map(([orderAttr, orderDirection]) => [
+    orderByAttribute(orderAttr, options),
+    orderDirection,
+  ]);
+  model.primaryKeyAttributes.forEach(primaryKeyAttribute => {
+    if (result.findIndex(([attr]) => attr === primaryKeyAttribute) < 0) {
+      result.push([primaryKeyAttribute, 'ASC']);
+    }
+  });
+  return result;
+}
+
+function reverseDirection(direction) {
+  return direction.indexOf('ASC') >= 0
+    ? direction.replace('ASC', 'DESC')
+    : direction.replace('DESC', 'ASC');
+}
+
+function reverseOrder(order) {
+  return order.map(([orderAttr, orderDirection]) => [orderAttr, reverseDirection(orderDirection)]);
+}
+
+
+/**
+ * Creates a cursor given a node returned from the Database
+ * @param  {Object}   node            sequelize model instance
+ * @param  {String[]} orderAttributes  the attributes pertaining in ordering
+ * @return {String}                   The Base64 encoded cursor string
+ */
+function toCursor(node, info) {
+  return base64(JSON.stringify(info.orderAttributes.map(attr => node.get(attr))));
```

Comment:
this makes it a migration-breaker,
any way we can make it so old cursors would still work?

Reply:
I'm confused what you mean, cursors are just opaque strings, and there's no way an old-style cursor would somehow pop up in the new code

Reply:
I mean unless users do something weird like persisting cursors until they upgrade, which we should discourage people from doing